deepchem/splits/splitters.py

```diff
@@ -698,60 +769,193 @@ def split(self,
       test_idx = np.hstack([test_idx, sortidx_split[shuffled[valid_cutoff:]]])
 
     # Append remaining examples to train
-    if sortidx.shape[0] > 0: np.hstack([train_idx, sortidx])
+    if sortidx.shape[0] > 0:
+      np.hstack([train_idx, sortidx])
 
     return (train_idx, valid_idx, test_idx)
 
 
+class IndexSplitter(Splitter):
+  """Class for simple order based splits.
+
+  Use this class when the `Dataset` you have is already ordered sa you would
+  like it to be processed. Then the first `frac_train` proportion is used for
+  training, the next `frac_valid` for validation, and the final `frac_test` for
+  testing. This class may make sense to use your `Dataset` is already time
+  ordered (for example).
+  """
+
+  def split(self,
+            dataset: Dataset,
+            frac_train: float = 0.8,
+            frac_valid: float = 0.1,
+            frac_test: float = 0.1,
+            seed: Optional[int] = None,
+            log_every_n: Optional[int] = None
+           ) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """Splits internal compounds into train/validation/test in provided order.
+
+    Parameters
+    ----------
+    dataset: Dataset
+      Dataset to be split.
+    frac_train: float, optional (default 0.8)
+      The fraction of data to be used for the training split.
+    frac_valid: float, optional (default 0.1)
+      The fraction of data to be used for the validation split.
+    frac_test: float, optional (default 0.1)
+      The fraction of data to be used for the test split.
+    seed: int, optional (default None)
+      Random seed to use.
+    log_every_n: int, optional
+      Log every n examples (not currently used).
+
+    Returns
+    -------
+    Tuple[np.ndarray, np.ndarray, np.ndarray]
+      A tuple of train indices, valid indices, and test indices.
+      Each indices is a numpy array.
+    """
+    np.testing.assert_almost_equal(frac_train + frac_valid + frac_test, 1.)
+    num_datapoints = len(dataset)
+    train_cutoff = int(frac_train * num_datapoints)
+    valid_cutoff = int((frac_train + frac_valid) * num_datapoints)
+    indices = range(num_datapoints)
+    return (indices[:train_cutoff], indices[train_cutoff:valid_cutoff],
+            indices[valid_cutoff:])
+
+
+class IndiceSplitter(Splitter):
+  """Split data in the fashion specified by user.
+
+  For some applications, you will already know how you'd like to split the
+  dataset. In this splitter, you simplify specify `valid_indices` and
+  `test_indices` and the datapoints at those indices are pulled out of the
+  dataset. Note that this is different from `IndexSplitter` which only splits
+  based on the existing dataset ordering, while this `IndiceSplitter` can
+  split on any specified ordering.
```

Comment:
Could you add a usage example here?

Reply:
What does the name mean?  "Indice" is not a word.  "Indices" is the plural of "index".

Reply:
Perhaps we should consider renaming `IndiceSplitter` to `SpecifiedSplitter`. This might solve the issue that we have examples using `SpecifiedSplitter` and solve the ambiguity with the names as well